### PR TITLE
[node] Add ClientRequest.prototype.path

### PR DESCRIPTION
**Pull request by gjbkz**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  1. https://nodejs.org/api/http.html#http_request_path
  2. https://github.com/nodejs/node/pull/25788

### CODE REVIEW

File: types/node/http.d.ts [hunk 1]
```diff
@@ -159,6 +159,7 @@ declare module "http" {
 
         constructor(url: string | URL | ClientRequestArgs, cb?: (res: IncomingMessage) => void);
 
+        path: string;
```
Comment (isLishude) [inline types/node/http.d.ts]: I think here should be `readonly path: string;`

File: types/node/test/http.ts [hunk 2]
```diff
@@ -112,6 +113,9 @@ import * as net from 'net';
 
     // event
     req.on('data', () => { });
+
+    // path
+    assert.equal(req.path.charAt(0), '/');
```
Comment (isLishude) [inline types/node/test/http.ts]: type tests don't need to assert,here should be `const path: string = req.path;`

**Review by islishude — CHANGES_REQUESTED** [review]

File: types/node/http.d.ts (hunk 1 above)
Comment (kei-ito) [reply types/node/http.d.ts]: Thank you for your suggestion. I've updated them.

File: types/node/test/http.ts (hunk 2 above)
Comment (kei-ito) [reply types/node/test/http.ts]: I misunderstood how the test works. I've committed the fix (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/33081/commits/4f17556377a38429be3c3fe456eb8ea9005079b4). Thank you.

Comment (isLishude) [reply types/node/test/http.ts]: you can look here for more info https://github.com/Microsoft/dtslint#write-tests

**Review by islishude — APPROVED** [review]

File: types/node/test/http.ts (hunk 2 above)
Comment (kei-ito) [reply types/node/test/http.ts]: I got it. Thank you for the link!

**Review by SimonSchick — APPROVED** [review]

**Review by BrunoScheufler — APPROVED** [review]

**Review by Flarna — APPROVED** [review]

**Review by r3nya — APPROVED** [review]

**Pull request merged**